cmd/minishift/cmd/root.go

```diff
@@ -333,3 +361,16 @@ func setDefaultActiveProfile() {
 		}
 	}
 }
+
+// checkProfileExist checks if a profile exist or not when --profile flag used.
+// If profile not exist then it will error out with message.
+func checkProfileExist(cmd *cobra.Command) {
+	if cmd.Parent() != nil {
+		// This condition true for each command execpt *m profile* and *m start ...*
+		if cmd.Parent().Name() != profileFlag && cmd.Name() != startFlag {
+			if !cmdUtil.IsValidProfile(constants.ProfileName) {
+				atexit.ExitWithMessage(1, fmt.Sprintf("Profile: %s doesn't exist, Use *minishift profile set %s* to create", constants.ProfileName, constants.ProfileName))
```

Comment:
Please update the error message to include `minishift start --profile PROFILE_NAME`. At this point we are only saying `minishift profile set`. E.g. `Use minishift start --profile or minishift profile set ` to create profile.